cmd/outline-ss-server/metrics.go

```diff
@@ -161,10 +277,32 @@ func (m *outlineMetrics) SetNumAccessKeys(numKeys int, ports int) {
 	m.ports.Set(float64(ports))
 }
 
-func (m *outlineMetrics) AddOpenTCPConnection(clientInfo ipinfo.IPInfo) {
+func (m *outlineMetrics) AddOpenTCPConnection(addr net.Addr) {
+	clientInfo, err := ipinfo.GetIPInfoFromAddr(m.IPInfoMap, addr)
```

Comment:
You are now looking ip the IP information multiple times. That is a very large map. Please revert to one lookup per connection.

This method doesn't even need the IP, so it can be completely reverted.